### PR TITLE
Todo/week 10

### PR DESCRIPTION
**Pull request by sghourab**
Included an example of custom progress views. 
https://drive.google.com/file/d/1TfCC8xpTLJ6SzMCAlAZCH2GEXa7fPhxJ/view?usp=share_link

### CODE REVIEW

**Comment by den-ree** [comment]
Screenshot 2022-12-16 at 15 37 28

@sghourab you need to sync with main branch before starting, looks like you have some local changes which triggers conflicts each time.

File: CourseManager.xcodeproj/xcuserdata/summercrow.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist [hunk 1]
```diff
@@ -36,5 +36,37 @@
             landmarkType = "24">
          </BreakpointContent>
       </BreakpointProxy>
+      <BreakpointProxy
```
Comment (den-ree) [inline CourseManager.xcodeproj/xcuserdata/summercrow.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:39]: @sghourab add please these files to `.gitignore`

File: CourseManager/Views/SampleListView.swift [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                    .fill(.ultraThinMaterial)
+                    .mask(RoundedRectangle(cornerRadius: 20, style: .continuous))
+            )
+            .padding(.horizontal, 20)
+            .padding(.vertical, 8)
+        }
+    }
+}
+
+
```
Comment (den-ree) [inline CourseManager/Views/SampleListView.swift:60]: @sghourab remove spaces

File: CourseManager/Views/ExampleViews/ProgressView/CircleProgressView.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+//  CircleProgressView.swift
+//  CourseManager
+//
+//  Created by Summer Crow on 2022-12-14.
+//
+
+import SwiftUI
+
+struct CircleProgressView: View {
+    var value: CGFloat = 0.95
```
Comment (den-ree) [inline CourseManager/Views/ExampleViews/ProgressView/CircleProgressView.swift:11]: @sghourab variables `value` and `percent` should be sync which each other. `percent` should be readonly variable and when we set `value` it should be updated.

File: CourseManager/Views/ExampleViews/MatchedGeometryEffect/Views/Components/FiveStarRatingView.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        .onAppear {
+            withAnimation(.spring().delay(0.7)) {
+                appear = true
+            }
+        }
+    }
+
+    private var starsView: some View {
+        HStack(spacing: 0) {
+            ForEach(/*@START_MENU_TOKEN@*/0 ..< 5/*@END_MENU_TOKEN@*/) { _ in
```
Comment (den-ree) [inline CourseManager/Views/ExampleViews/MatchedGeometryEffect/Views/Components/FiveStarRatingView.swift:30]: remove hidden code

**Review by den-ree — CHANGES_REQUESTED** [review]
@sghourab check comments and fix conflicts

File: CourseManager/Views/ExampleViews/MatchedGeometryEffect/Views/Components/FiveStarRatingView.swift (hunk 4 above)
Comment (sghourab) [reply CourseManager/Views/ExampleViews/MatchedGeometryEffect/Views/Components/FiveStarRatingView.swift:30]: Not sure why it's showing like that, in my code it's actually:
  ForEach(0 ..< 5) { _ in
                Image(systemName: "star.fill")
                    .foregroundColor(.gray)
            }

File: CourseManager.xcodeproj/xcuserdata/summercrow.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist (hunk 1 above)
Comment (sghourab) [reply CourseManager.xcodeproj/xcuserdata/summercrow.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:39]: How would I find these files?

Comment (sghourab) [reply CourseManager.xcodeproj/xcuserdata/summercrow.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:39]: Is this how you would do it? I read online that i need to temporarily delete file, then commit and re add it. If so, I don't know how to access it from xcode. Should delete it from finder instead?

Screenshot 2022-12-20 at 9 56 29 AM

**Pull request merged**